coderd/database/databasefake/databasefake.go

```diff
@@ -1717,6 +1717,46 @@ func (q *fakeQuerier) GetTemplateVersionByJobID(_ context.Context, jobID uuid.UU
 	return database.TemplateVersion{}, sql.ErrNoRows
 }
 
+func (q *fakeQuerier) GetPreviousTemplateVersion(_ context.Context, arg database.GetPreviousTemplateVersionParams) (database.TemplateVersion, error) {
+	q.mutex.RLock()
+	defer q.mutex.RUnlock()
+
+	var currentTemplateVersion database.TemplateVersion
+	for _, templateVersion := range q.templateVersions {
+		if templateVersion.TemplateID != arg.TemplateID {
+			continue
+		}
+		if templateVersion.Name != arg.Name {
+			continue
+		}
+		if templateVersion.OrganizationID != arg.OrganizationID {
+			continue
+		}
+		currentTemplateVersion = templateVersion
```

Comment:
How does this work? Is it relying on the order of `q.templateVersions`? How about combining this loop and the previous one to create a slice of relevant template versions, sorting it, and picking the one before the newest?

Reply:
>How does this work? 

I tried to emulate the SQL query logic.

> Is it relying on the order of q.templateVersions?

Yeap.

> How about combining this loop and the previous one to create a slice of relevant template versions, sorting it, and picking the one before the newest?

Yeah, I tried that but the code doesn't look pretty easy to follow IMO. I know it is not the best performant way but I don't see a problem since it is only for tests but I can do it if you think it is better.